File: interview-coach-frontend/src/types/career.ts
export type CurrentLevel = "Beginner" | "Intermediate" | "Advanced";

export interface WeeklyPlanItem {
  week: number;
  focus: string;
  tasks: string[];
}

export interface CareerRoadmapData {
  current_level: CurrentLevel;
  strengths: string[];
  weaknesses: string[];
  weekly_plan: WeeklyPlanItem[];
  recommended_projects: string[];
  recommended_leetcode_topics: string[];
  recommended_interview_topics: string[];
  recommended_courses: string[];
}

export interface GenerateRoadmapResponse extends CareerRoadmapData {
  id: string;
  created_at: string;
}

export interface RoadmapSummary {
  id: string;
  current_level: CurrentLevel;
  created_at: string;
}

export interface RoadmapHistoryResponse {
  total: number;
  items: RoadmapSummary[];
}

File: interview-coach-backend/schemas/career.py
from datetime import datetime
from typing import Literal
from pydantic import BaseModel, Field


CurrentLevel = Literal["Beginner", "Intermediate", "Advanced"]


class WeeklyPlanItem(BaseModel):
    week: int
    focus: str
    tasks: list[str] = Field(default_factory=list)


class CareerRoadmapData(BaseModel):
    current_level: CurrentLevel
    strengths: list[str] = Field(default_factory=list)
    weaknesses: list[str] = Field(default_factory=list)
    weekly_plan: list[WeeklyPlanItem] = Field(default_factory=list)
    recommended_projects: list[str] = Field(default_factory=list)
    recommended_leetcode_topics: list[str] = Field(default_factory=list)
    recommended_interview_topics: list[str] = Field(default_factory=list)
    recommended_courses: list[str] = Field(default_factory=list)


class GenerateRoadmapResponse(CareerRoadmapData):
    id: str
    created_at: datetime


class RoadmapSummary(BaseModel):
    id: str
    current_level: CurrentLevel
    created_at: datetime

    class Config:
        from_attributes = True


class RoadmapHistoryResponse(BaseModel):
    total: int
    items: list[RoadmapSummary]